config/config.go

```diff
@@ -234,6 +238,14 @@ func (cfg Config) Validate() error {
 		}
 	}
 
+	for _, filesystem := range cfg.Filesystems {
```

Comment:
I'm not a fan of separating this validation out. Some Go-fu help here would be nice to pull this up into the `validators`.

Reply:
Ahh, I think we need to define `Validate` and `Copy` on `[]*Filesystem` structs, which then would call the `Validate` and `Copy` funcs on `Filesystem` structs when necessary

Then you can move the `cfg.Filesystems` into the above validators